File: bms/go/geo/geo_s1_Interval_Contains/geo_s1_Interval_Contains.go
package test

import (
	"math"
)

func (i geo_s1_Interval) Contains(p float64) bool {
	if p == -math.Pi {
		p = math.Pi
	}
	return i.fastContains(p)
}

func (i geo_s1_Interval) fastContains(p float64) bool {
	if i.IsInverted() {
		return (p >= i.Lo || p <= i.Hi) && !i.IsEmpty()
	}
	return p >= i.Lo && p <= i.Hi
}

func (i geo_s1_Interval) IsInverted() bool { return i.Lo > i.Hi }

func (i geo_s1_Interval) IsEmpty() bool { return i.Lo == math.Pi && i.Hi == -math.Pi }

type geo_s1_Interval struct {
	Lo, Hi float64
}

File: translations_only/success/go/geo/geo_s1_Interval_Contains/gpt4/BaseRepair/geo_s1_Interval_Contains.rs

use std::f64::consts::PI;

struct GeoS1Interval {
    lo: f64,
    hi: f64,
}

impl GeoS1Interval {
    fn contains(&self, mut p: f64) -> bool {
        if p == -PI {
            p = PI;
        }
        self.fast_contains(p)
    }

    fn fast_contains(&self, p: f64) -> bool {
        if self.is_inverted() {
            return (p >= self.lo || p <= self.hi) && !self.is_empty();
        }
        p >= self.lo && p <= self.hi
    }

    fn is_inverted(&self) -> bool {
        self.lo > self.hi
    }

    fn is_empty(&self) -> bool {
        self.lo == PI && self.hi == -PI
    }
}
